app/js/team.js

```diff
@@ -72,125 +91,244 @@ angular.module('teamform-team-app', ['firebase'])
-		if ( index > -1 ) {
+		if ( index > -1 ) 
+		{
 			$scope.param.teamMembers.splice(index, 1); // remove that item
+			$scope.param.currentTeamSize --;
+			$scope.saveFunc();
+		}
+	}
+	
+	$scope.changeLeader = function(leaderToBeSwapped = null, memberToBeSwapped = null) 
+	{
+		var indexLeader = $scope.param.teamLeaders.indexOf(leaderToBeSwapped);
+		var indexMember = $scope.param.teamMembers.indexOf(memberToBeSwapped);
+
+		if ( indexLeader > -1 && indexMember > -1) {
+			$scope.param.teamLeaders.splice(indexLeader, 1);
+			$scope.param.teamMembers.splice(indexMember, 1);
+			$scope.param.teamLeaders.push(memberToBeSwapped);
+			$scope.param.teamMembers.push(leaderToBeSwapped);
+			
+			$scope.saveFunc();
+		}
+	}
+
+	$scope.addLeader = function(member = null) 
+	{
+		var indexMember = $scope.param.teamMembers.indexOf(member);
+		if ( indexMember > -1 )
+		{
+			$scope.param.teamMembers.splice(indexMember, 1);
+			$scope.param.teamLeaders.push(member);
+			$scope.param.currentTeamLeaderSize++;
+			
+			$scope.saveFunc();
+		}
+	}
+	
+	$scope.removeLeader = function(leader = null) 
+	{
+		var indexLeader = $scope.param.teamLeaders.indexOf(leader);
+		if ( indexLeader > -1 )
+		{
+			$scope.param.teamLeaders.splice(indexLeader, 1);
+			$scope.param.teamMembers.push(leader);
+			$scope.param.currentTeamLeaderSize--;
 			
 			$scope.saveFunc();
 		}
-		
 	}
 	
+	$scope.addWantedSkill = function(skillString = null) 
+	{
+			$scope.addString(skillString, $scope.param.wantedSkills);
+	}
 	
+	$scope.addWantedPersonalities = function(personalityString = null) 
+	{
+			$scope.addString(personalityString, $scope.param.wantedPersonalities);
+	}
 	
+	$scope.addWantedHoroscopes = function(horoscopeString = null) 
+	{
+			$scope.addString(horoscopeString, $scope.param.wantedHoroscopes);
+	}
 	
+	$scope.addString = function(stringToBeAdded, parameter)
+	{
+		var indexString = parameter.indexOf(stringToBeAdded);
+		if (indexString < 0)
+		{
+			parameter.push(stringToBeAdded);
+			$scope.saveFunc();
+		}
+	}
 	
+	$scope.removeWantedSkill = function(skillString = null) 
+	{
+			$scope.removeString(skillString, $scope.param.wantedSkills);
+	}
 	
-		
-}]);
\ No newline at end of file
+	$scope.removeWantedPersonalities = function(personalityString = null) 
+	{
+			$scope.removeString(personalityString, $scope.param.wantedPersonalities);
+	}
+	
+	$scope.removeWantedHoroscopes = function(horoscopeString = null) 
+	{
+			$scope.removeString(horoscopeString, $scope.param.wantedHoroscopes);
+	}
+	
+	$scope.removeString = function(stringToBeRemoved, parameter)
+	{
+		var indexString = parameter.indexOf(stringToBeRemoved);
+		if (indexString > -1)
+		{
+			parameter.splice(indexString, 1);
+			$scope.saveFunc();
+		}
+	} 
+	
+	$scope.calculateNumPrettyGirls = function()
+	{
+		int count = 0;
+		for(int i = 0; i < $scope.param.teamMembers.length; i++)
+		{
+			if($scope.param.teamMembers[i].gender == 'F' || $scope.param.teamMembers[i].gender == 'f')
```

Comment:
teamMembers will not contain information
If you need his/her gender, you should get it from profile